src/frontend/context/MapStyleContext.tsx

```diff
@@ -25,16 +47,49 @@ export const MapStyleContext: React.Context<MapStyleContextType> = React.createC
 const usePersistedState = createPersistedState(MAP_STYLE_KEY);
 
 export const MapStyleProvider: React.FC = ({ children }) => {
-  const [styleId, , setStyleId] = usePersistedState<string | null>(null);
+  const [mapServerReady, setMapServerReady] = React.useState(false);
+  const [styleId, status, setStyleId] = usePersistedState<string>("");
+  const [{ backgroundMaps }] = useExperiments();
+
+  const [onlineMapState, setOnlineState] = React.useState<OnlineState>(
+    "unknown"
+  );
 
-  const contextValue = React.useMemo(() => [styleId, setStyleId] as const, [
-    styleId,
-    setStyleId,
-  ]);
+  // TODO: Eventually use the net info module to determine if internet access is available
+  // Currently not truly a reflection of internet access
+  React.useEffect(() => {
+    let didCancel = false;
+
+    ky.get(normalizeStyleURL(onlineStyleURL, config.mapboxAccessToken))
+      .json()
+      .then(() => didCancel || setOnlineState("online"))
+      .catch(() => didCancel || setOnlineState("offline"));
+
+    return () => {
+      didCancel = true;
+    };
+  }, []);
+
+  React.useEffect(() => {
+    let subscription: Subscription | undefined;
+
+    if (backgroundMaps) {
+      subscription = api.maps.addServerStateListener(({ value }) => {
+        setMapServerReady(value === "started");
```

Comment:
nit: not sure how frequently the listener fires, but maybe use the functional update here.

```suggestion
        setMapServerReady(prev => {
          const next = value === "started"
          return prev === next ? prev : next
        );
```

probably not too much of an issue for now